src/cli/commands/install.js

```diff
@@ -232,6 +232,40 @@ export class Install {
       const json = await this.config.readJson(loc);
       await normalizeManifest(json, this.config.cwd, this.config, true);
 
+      const aggregateManifests = async (root: Manifest, rootCwd: string) => {
+        if (root.workspaces) {
+          for (const workspace of root.workspaces) {
+            let workspaceCwd = path.join(rootCwd, workspace);
+            const workspaceLoc = path.join(workspaceCwd, filename);
+            if (!(await fs.exists(workspaceLoc))) {
+              continue;
+            }
+            const workspaceJson: Manifest = await this.config.readJson(workspaceLoc);
+            await normalizeManifest(workspaceJson, workspaceCwd, this.config, true);
+
+            // merge deps
+            for (const type of ['dependencies', 'devDependencies', 'optionalDependencies']) {
+              if (workspaceJson[type]) {
+                for (const key in workspaceJson[type]) {
+                  if (root[type] && root[type][key] && root[type][key] !== workspaceJson[type][key]) {
+                    this.reporter.warn(
+                      this.reporter.lang('incompatibleDependenciesInWorkspace', key, workspaceCwd, rootCwd),
+                    );
```

Comment:
> When conflict arise, prevent hoisting the conflicting modules (so that they end up being installed into their respective node_modules folders)

Problem being that in such a case, we would have to run `yarn install` recursively into the workspace directories